src/BuildScriptGenerator/PreAndPostCommandScriptHelper.cs

```diff
@@ -0,0 +1,74 @@
+﻿// --------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// --------------------------------------------------------------------------------------------
+
+using System.IO;
+
+namespace Microsoft.Oryx.BuildScriptGenerator
+{
+    internal static class PreAndPostBuildCommandHelper
+    {
+        public static (string preBuildCommand, string postBuildCommand) GetPreAndPostBuildCommands(
```

Comment:
suggestion: move this to be a method of `EnvironmentSettings`?
Or even better, split the code of this method to be getters of `EnvironmentSettings.PreBuildCommand` & `EnvironmentSettings.PostBuildCommand`

Reply:
Yeah, I considered it initially but it felt like having stuff like quotes around the command or not(depending on whether the content is a file or not) seem to be a wrong place/layer for EnvironmentSettings or DefaultEnvironmentSettingsProvider as it is related to how shell script works. Like these apis only give the input (yeah they sanitize a bit, like trimming space and quotes).

Reply:
guiding question - will anyone ever use `EnvironmentSettings.PreBuildCommand`/`EnvironmentSettings.PostBuildCommand` **without** `GetPreAndPostBuildCommands`?

Reply:
Yes, they could...are you implying they wouldn't be getting the logic that is present in `GetPreAndPostBuildCommands`? yes, they would not...but building that logic into those properties seems like a wrong layer as by doing `EnvironmentSettings.PreBuildCommand` you are just trying to get the value supplied by the user (enclosing with quotes or not is specific to the script IMO)

Reply:
ok, got it 👍